lib/rexml/parsers/xpathparser.rb

```diff
@@ -363,6 +363,18 @@ def Predicate path, parsed
         path
       end
 
+      def QuoteLiteral literal
+        case literal
+        when String
+          # Xpath 1.0 does not support escape characters.
+          # Assumes literal does not contain both single and double quotes.
+          pattern = literal.include?('"') ? "'%s'" : '"%s"'
+          pattern % literal
```

Comment:
Could you use string interpolation instead of `String#%`?

```suggestion
          if literal.include?("'")
            "\"#{literal}\""
          else
            "'#{literal}'"
          end
```